### PR TITLE
Don't call `init_site()` when `lazy = TRUE` + use posix line end + document `init_site()` + speed cli console output 

### PR DESCRIPTION
**Pull request by olivroy**
I am not talking about `init_site()` anymore after this! 

Follow-up to #2571 

Only call `init_site()` if `lazy = FALSE`

Document where `init_site()` is actually needed. https://github.com/r-lib/pkgdown/pull/2557#issuecomment-2121017718

To speed console printing, create `cache_cli_colours()`.  (bench marks in https://github.com/r-lib/pkgdown/pull/2557#issuecomment-2153074189)  Tip taken from r-lib/cli#607

I put in the Posix line end thing as a follow-up to https://github.com/r-lib/testthat/issues/1958#issuecomment-2152596705

I am no longer seeing spurious diff in snapshot tests this way.

Still haven't figure out no-pandoc test failure! Will take a look again after #2645. This appears to be stochastic (just encountered it on main)

### CODE REVIEW

File: R/context.R [hunk 1]
```diff
@@ -4,10 +4,18 @@ section_init <- function(pkg, depth, override = list(), .frame = parent.frame())
...
   local_envvar_pkgdown(pkg, .frame)
   local_options_link(pkg, depth = depth, .frame = .frame)
+  local_options_console(.frame = .frame)
 
   pkg
 }
 
+local_options_console <- function(.frame = parent.frame()) {
+  # Improves speed of init_site() with tip from r-lib/cli#607
+  # Only does so if no option set
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/context.R:14]: Changing the name of the function makes the purpose more obvious, and it's easier to click on a link to get the details.

**Review by hadley — APPROVED** [review]
Looks good!

File: R/context.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (olivroy) [reply R/context.R]: FWIW, in RStudio you can click on things like that :)  https://github.com/rstudio/rstudio/issues/11840

File: R/init.R [hunk 2]
```diff
@@ -21,6 +29,7 @@
 #' @inheritParams build_articles
 #' @export
 init_site <- function(pkg = ".") {
+  local_options_console()
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/init.R]: Now clear that it shouldn't be called here 😄

[the remark this answers is not in the document]
Comment (olivroy) [reply R/init.R]: Hi Hadley, actually, `cache_cli_colours()` should be used in `init_site()`,

bench marks

```r
# without caching
 expression    min median `itr/sec` mem_alloc `gc/sec` n_itr  n_gc total_time result
  <bch:expr>  <bch> <bch:>     <dbl> <bch:byt>    <dbl> <int> <dbl>   <bch:tm> <list>
1 init_site() 3.23s  3.23s     0.309    14.7MB    0.619     1     2      3.23s <NULL>
# with caching
  expression    min median `itr/sec` mem_alloc `gc/sec` n_itr  n_gc total_time result
  <bch:expr>  <bch> <bch:>     <dbl> <bch:byt>    <dbl> <int> <dbl>   <bch:tm> <list>
1 init_site() 3.02s  3.02s     0.331    8.89MB    0.992     1     3      3.02s <NULL>
```

Significant difference in memory usage. And also, the output prints much faster in the console!

File: pkgdown.Rproj
Comment (olivroy) [inline pkgdown.Rproj:1]: Sorry about this one. I am still struggling with line ending.. I solved the snapshot part, but apprently .Rproj is the new problem https://github.com/rstudio/rstudio/issues/1929 :(

Comment (olivroy) [reply pkgdown.Rproj:1]: Made the commit from my mac, and it worked!

**Review by hadley — APPROVED** [review]

**Pull request merged**